contracts/vaults/YearnVaults/base/RibbonVault.sol

```diff
@@ -0,0 +1,744 @@
+
+        (uint256 lockedBalance, uint256 newPricePerShare, uint256 mintShares) =
+            VaultLifecycleYearn.rollover(
+                totalSupply(),
+                totalBalance(),
+                vaultParams,
+                vaultState
+            );
+
+        optionState.currentOption = newOption;
+        optionState.nextOption = address(0);
+
+        // Finalize the pricePerShare at the end of the round
+        uint16 currentRound = vaultState.round;
+        roundPricePerShare[currentRound] = newPricePerShare;
+
+        // Take management / performance fee from previous round and deduct
+        lockedBalance = lockedBalance.sub(_collectVaultFees(lockedBalance));
+
+        vaultState.totalPending = 0;
+        vaultState.round = currentRound + 1;
+
+        _mint(address(this), mintShares);
+
+        // Wrap entire `asset` balance to `collateralToken` balance
+        VaultLifecycleYearn.wrapToYieldToken(
+            vaultParams.asset,
+            address(collateralToken)
+        );
+
+        return (newOption, lockedBalance);
+    }
+
+    /*
+     * @notice Helper function that transfers management fees and performance fees from previous round.
+     * @param currentLockedBalance is the balance we are about to lock for next round
+     * @return vaultFee is the fee deducted
+     */
+    function _collectVaultFees(uint256 currentLockedBalance)
+        internal
+        returns (uint256 vaultFee)
+    {
+        uint256 prevLockedAmount = vaultState.lastLockedAmount;
+
+        // Take performance fee and management fee ONLY if difference between
+        // last week and this week's vault deposits, taking into account pending
+        // deposits and withdrawals, is positive. If it is negative, last week's
+        // option expired ITM past breakeven, and the vault took a loss so we
+        // do not collect performance fee for last week
+        if (
+            currentLockedBalance.sub(vaultState.totalPending) > prevLockedAmount
+        ) {
+            uint256 performanceFeeInAsset =
+                performanceFee > 0
+                    ? currentLockedBalance
+                        .sub(vaultState.totalPending)
+                        .sub(prevLockedAmount)
+                        .mul(performanceFee)
+                        .div(100 * 10**6)
+                    : 0;
+            uint256 managementFeeInAsset =
+                managementFee > 0
+                    ? currentLockedBalance.mul(managementFee).div(100 * 10**6)
+                    : 0;
+
+            vaultFee = performanceFeeInAsset.add(managementFeeInAsset);
+        }
+
+        if (vaultFee > 0) {
+            VaultLifecycleYearn.withdrawYieldAndBaseToken(
+                WETH,
+                vaultParams.asset,
+                address(collateralToken),
+                feeRecipient,
+                vaultFee
+            );
+            emit CollectVaultFees(performanceFee, vaultFee, vaultState.round);
```

Comment:
Refer to https://github.com/ribbon-finance/ribbon-v2/pull/47/files#r676746116

`emit CollectVaultFees(performanceFee, managementFee, vaultState.round);`

Reply:
see above